src/main/scala/chisel3/util/experimental/pla.scala

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.experimental
+
+import chisel3._
+import chisel3.util.{BitPat, Cat}
+
+object pla {
+
+  /** Construct a [[https://en.wikipedia.org/wiki/Programmable_logic_array]] from specified table.
+    * @param table A [[Seq]] of inputs -> outputs mapping
+    *              Each position in the input matrix corresponds to an input variable where a `0` implies the corresponding
+    *              input literal appears complemented in the product term, a `1` implies the input literal appears un-
+    *              complemented in the product term, and `?` implies the input literal does not appear in the product term.
+    *              For each output, a `1` means this product term makes the function value a `1`, and a `0` or `?` means
+    *              this product term has no meaning for the value of this function.
+    * @param invert A [[BitPat]] specify which bit of the output should be inverted. A `1` means the correspond position
+    *               of the output should be inverted in the PLA, a `0` or a `?` means direct output from the OR matrix.
+    * @return the input [[Wire]] of [[UInt]] of the constructed pla.
+    * @return the output [[Wire]] of [[UInt]] of the constructed pla.
+    * @example A 1-of-8 decoder (like the 74xx138) can be constructed as follow
+    *          {{{
+    *            val (inputs, outputs) = pla(Seq(
+    *              (BitPat("b000"), BitPat("b00000001")),
+    *              (BitPat("b001"), BitPat("b00000010")),
+    *              (BitPat("b010"), BitPat("b00000100")),
+    *              (BitPat("b011"), BitPat("b00001000")),
+    *              (BitPat("b100"), BitPat("b00010000")),
+    *              (BitPat("b101"), BitPat("b00100000")),
+    *              (BitPat("b110"), BitPat("b01000000")),
+    *              (BitPat("b111"), BitPat("b10000000")),
+    *            ))
+    *          }}}
+    */
```

Comment:
You may want to run this through Scaladoc to see what it looks like. The length of the `@param` descriptions seem too long and may render weirdly. Also, I think `@example` is discouraged now in favor of just prose with `{{{...}}}` blocks.

Reply:
Got it I will check Scaladoc here!